modules/images/dominant-color/load.php

```diff
@@ -339,3 +320,48 @@ function wp_get_attachment_file_path( $attachment_id, $size = 'medium' ) {
 
 	return $filepath;
 }
+
+/**
+ * Gets the Dominant color for an image attachment.
+ *
+ * @since n.e.x.t
+ *
+ * @param int $attachment_id Attachment ID for image.
+ * @return string|null|WP_Error hex value of dominant color or null if not set.
+ */
+function dominant_color_get_dominant_color( $attachment_id ) {
+	if ( ! wp_attachment_is_image( $attachment_id ) ) {
+		return null;
+	}
+	$image_meta = wp_get_attachment_metadata( $attachment_id );
+	if ( ! is_array( $image_meta ) ) {
+		return new WP_Error( 'unable_to_find_imagemeta_for_id', __( 'Unable to find metadata for image for given ID', 'performance-lab' ) );
```

Comment:
Functions that just return existing data shouldn't return a `WP_Error`, but rather `null`. If there is no image metadata, this is not an error for this function, it just means no dominant color is set.
```suggestion
		return null;
```